tools/grub/Dockerfile

```diff
@@ -0,0 +1,69 @@
+FROM linuxkit/alpine:3683c9a66cd4da40bd7d6c7da599b2dcd738b559 AS grub-build
+RUN apk add \
+  automake \
+  make \
+  bison \
+  coreutils \
+  gettext \
+  gettext-dev \
+  flex \
+  gcc \
+  git \
+  libtool \
+  libc-dev \
+  linux-headers \
+  patch \
+  pkgconfig \
+  python3 \
+  autoconf
+
+# because python is not available
+RUN ln -s python3 /usr/bin/python
+
+ENV GRUB_MODULES="ext2 iso9660 gzio linux acpi normal cpio crypto disk boot crc64 part_gpt tftp xzio xfs video"
+ENV GRUB_COMMIT=5bc41db756c5b342aa22fc95508bde105e05a095
+ENV SRCDIR=/src
+
+# separate steps for better caching
+RUN mkdir /grub-lib ${SRCDIR} && \
+  set -e && \
+  cd ${SRCDIR} && \
+  git clone git://git.savannah.gnu.org/grub.git && \
+  cd grub && \
+  git checkout -b grub-build ${GRUB_COMMIT}
+
+WORKDIR ${SRCDIR}/grub
+
+# this is long and slow, so keep separate step
+RUN ./bootstrap
+
+# run these in separate steps for caching. 
+# determine uname -m in each step, since we want to replace it with a single cross-build when we get it working
```

Comment:
we don't generally (at present) cross build anything, so I am not sure it makes sense to have a single cross built package in future...

Reply:
In general, I would agree. But the net result of this whole thing isn't a traditional package as much as a single binary per platform, `/BOOTX64.EFI` and `/BOOTAA64.EFI` (to start). It would be nice to run this once and get all of the outputs. In the meantime, doesn't hurt to leave it, does it?

Reply:
That is a very good point... and a design shortcoming. If you consider that an OCI image can be useful not only to run things but also to package things without running them, it would be nice to have it be able to. Oh well, work with what we have.

Reply:
Well, I will admit it makes for a cleaner Dockerfile...